tutorials/SingleQubitSystemMeasurements/Tests.qs

```diff
@@ -0,0 +1,179 @@
+                    Adjoint statePrep(q, state);
+                    AssertQubit(Zero, q);
+                } else {
+                    Reset(q);
+                }
+            }
+        }
+        
+        mutable totalMisclassifications = 0;
+        for (i in 0 .. nStates - 1) {
+            if (misclassifications[i] != 0) {
+                set totalMisclassifications += misclassifications[i];
+                Message($"Misclassified {stateName[i]} as {stateName[1 - i]} in {misclassifications[i]} test runs.");   
+            }
+        }
+        
+        // This check will tell the total number of failed classifications
+        Fact(totalMisclassifications == 0, $"{totalMisclassifications} test runs out of {nTotal} returned incorrect state (see output for details).");
+    }
+
+    // ------------------------------------------------------
+    // Exercise 2. Distinguish |0❭ and |1❭
+    // ------------------------------------------------------
+    operation StatePrep_IsQubitZero (q : Qubit, state : Int) : Unit is Adj {
+        if (state == 0) {
+            // convert |0⟩ to |1⟩
+            X(q);
+        }
+    }
+
+    @Test("Microsoft.Quantum.Katas.CounterSimulator")
+    operation T2_IsQubitZero () : Unit {
+        DistinguishTwoStates(StatePrep_IsQubitZero, IsQubitZero, ["|1⟩", "|0⟩"], false);
+    }
+
+
+
+    // ------------------------------------------------------
+    // Exercise 3. Distinguish |+❭ and |-❭ using Measure operation
+    // ------------------------------------------------------
+    operation StatePrep_IsQubitMinus (q : Qubit, state : Int) : Unit is Adj {
+        if (state == 1) {
+            // convert |0⟩ to |-⟩
+            X(q);
+            H(q);
+        } else {
+            // convert |0⟩ to |+⟩
+            H(q);
+        }
+    }
+
+    @Test("Microsoft.Quantum.Katas.CounterSimulator")
+    operation T3_IsQubitMinus () : Unit {
+        DistinguishTwoStates(StatePrep_IsQubitMinus, IsQubitMinus, ["|+⟩", "|-⟩"], false);
+    }
+
+    // ------------------------------------------------------
+    // Exercise 5. Distinguishing specific orthogonal states in Q#
+    // ------------------------------------------------------
+
+    // |\psi + ⟩ =   0.6 * |0⟩ + 0.8 * |1⟩,
+    // |\psi + ⟩ =   -0.8 * |0⟩ + 0.6 * |1⟩.
+    operation StatePrep_IsQubitPsiPlus (q : Qubit, state : Int) : Unit is Adj {
+        if (state == 0) {
+            // convert |0⟩ to |psi -⟩
+            X(q);
+            Ry(2.0 * ArcTan2(0.8, 0.6), q);
+        } else {
+            // convert |0⟩ to |psi +⟩
+            Ry(2.0 * ArcTan2(0.8, 0.6), q);
+        }
+    }
+
+    @Test("Microsoft.Quantum.Katas.CounterSimulator")
+    operation T5_IsQubitSpecificState () : Unit {
+        // cross-test
+        // alpha = 0.0 or PI() => !isQubitOne
+        // alpha = PI() / 2.0 => isQubitOne
+        DistinguishTwoStates(StatePrep_IsQubitPsiPlus, IsQubitSpecificState, 
+            ["|psi-⟩", "|psi+⟩"], false);
```

Comment:
You can use Unicode ψ₋ and ψ₊ here to be consistent with the task description, as long as they are not distorted by Jupyter Notebook output